This file is part of MXE.
See index.html for further information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quintus <quintus@quintilianus.eu>
Date: Fri, 29 Jul 2016 23:11:19 +1000
Subject: [PATCH] Fix linking order.

Fixed upstream since commit bf699651740a1206c9ce214b91426db1e709ddc4,
which was after version 0.8.7, so remove this patch when upgrading to
version 0.8.8 or newer.
See: https://bitbucket.org/cegui/cegui/issues/1131/opengl-renderers-fail-to-crosscompile-due

diff --git a/cegui/src/RendererModules/OpenGL/CMakeLists.txt b/cegui/src/RendererModules/OpenGL/CMakeLists.txt
index 1111111..2222222 100644
--- a/cegui/src/RendererModules/OpenGL/CMakeLists.txt
+++ b/cegui/src/RendererModules/OpenGL/CMakeLists.txt
@@ -52,9 +52,9 @@ if (${CEGUI_USE_EPOXY})
     cegui_add_dependency(${CEGUI_TARGET_NAME} GLM)
     cegui_add_dependency(${CEGUI_TARGET_NAME} EPOXY)
 elseif (${CEGUI_USE_GLEW})
-    cegui_add_dependency(${CEGUI_TARGET_NAME} OPENGL)
     cegui_add_dependency(${CEGUI_TARGET_NAME} GLM)
     cegui_add_dependency(${CEGUI_TARGET_NAME} GLEW PUBLIC)
+    cegui_add_dependency(${CEGUI_TARGET_NAME} OPENGL)
 endif ()
 
 cegui_target_link_libraries(${CEGUI_TARGET_NAME} ${CEGUI_BASE_LIBNAME})

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quintus <quintus@quintilianus.eu>
Date: Fri, 29 Jul 2016 23:12:27 +1000
Subject: [PATCH] Build .pc files on Windows

CEGUI does not build the .pc files when targetting Windows.
See upstream ticket: https://bitbucket.org/cegui/cegui/issues/1135

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1111111..2222222 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -599,7 +599,7 @@ configure_file( cegui/include/CEGUI/ModuleConfig.h.in cegui/include/CEGUI/Module
 configure_file( samples/common/include/CEGUISamplesConfig.h.in samples/common/include/CEGUISamplesConfig.h )
 configure_file( doc/doxygen/doxyfile.in doc/doxygen/doxyfile )
 
-if (UNIX AND NOT APPLE)
+if (NOT(APPLE))
     configure_file( cegui/CEGUI.pc.in cegui/CEGUI-${CEGUI_VERSION_MAJOR}.pc @ONLY )
     install(FILES ${PROJECT_BINARY_DIR}/cegui/CEGUI-${CEGUI_VERSION_MAJOR}.pc DESTINATION ${CEGUI_PKGCONFIG_INSTALL_DIR})
 
